com.unity.formats.fbx/Tests/FbxTests/ModelExporterTest.cs

```diff
@@ -712,53 +717,72 @@ public class SkinnedMeshTestDataClass
             var expVerts = exportedSkinnedMesh.sharedMesh.vertices;
             Assert.That (expVerts, Is.Not.Null);
 
-            var origBoneWeights = originalSkinnedMesh.sharedMesh.boneWeights;
-            Assert.That (origBoneWeights, Is.Not.Null);
-            Assert.That (origBoneWeights.Length, Is.GreaterThan (0));
-
-            var expBoneWeights = exportedSkinnedMesh.sharedMesh.boneWeights;
-            Assert.That (expBoneWeights, Is.Not.Null);
-            Assert.That (expBoneWeights.Length, Is.GreaterThan (0));
-
             var origBones = originalSkinnedMesh.bones;
             var expBones = exportedSkinnedMesh.bones;
 
-            int comparisonCount = 0;
-            int minVertCount = Mathf.Min (origVerts.Length, expVerts.Length);
-            for(int i = 0; i < minVertCount; i++){
-                for (int j = 0; j < minVertCount; j++) {
-                    if (origVerts [i] == expVerts [j]) {
-                        // compare bone weights
-                        var origBw = origBoneWeights[i];
-                        var expBw = expBoneWeights [j];
+            // test with more than 4 bone weights
+
+            // Get the number of bone weights per vertex
+            var origBonesPerVertex = origMesh.GetBonesPerVertex();
+            var exportedBonesPerVertex = exportedMesh.GetBonesPerVertex();
+
+            // Get all the bone weights, in vertex index order
+            var origBoneWeights1 = origMesh.GetAllBoneWeights();
+            var exportedBoneWeights1 = exportedMesh.GetAllBoneWeights();
+
+            // get the vertex index to index in bone weights mapping (for easier access)
+            var origVertToBoneWeightIndex = new Dictionary<int, int>();
```

Comment:
this and `exportedVertToBoneWeightIndex` could be arrays I think?

Reply:
yep